### ISSUE TITLE
Announce once a language is automatically detected

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#129698]
1. Untitled file
2. Start typing
3. Once a language is automatically selected we should announce that, such that screen readers are aware that we have automatically selected a language

The message should be concise, something like "Python Language Mode is Set"

Also when "Auto Detect" is triggered from the Select Language Mode. But we should announce only if the mode changed.

**Comment by TylerLeonhardt** [comment]
Was "Auto Detect" done by @bpasero?

**Comment by isidorn** [comment]
Probably.

**Comment by TylerLeonhardt** [comment]
Ok, I'll wait for him to come back. I'd like to do this generically for all things that change the mode. I'm even thinking about Notebooks influencing the editor language based on the kernel.

**Comment by isidorn** [comment]
Yeah, having a general solution makes good sense.

**Comment by TylerLeonhardt** [comment]
@isidorn do you have any thoughts on where the best place for this announcement might live?

**Comment by isidorn** [comment]
I would probably do it as soon as you change the `mode` of the `Editor`.
It should not live in your service, since the service is just guessing, it know nothing about editors and modes. So I think clients of the service should take care of it.

**Comment by TylerLeonhardt** [comment]
Yeah sorry I guess I was wondering if there was a particular service for announcing arbitrary text. Then maybe I can put that in whatever service handles rendering of an editor?

**Comment by isidorn** [comment]
@TylerLeonhardt there is no service, there are just helper methods [`alert`](https://github.com/microsoft/vscode/blob/main/src/vs/base/browser/ui/aria/aria.ts#L49). 

I do not think this should be in a service that renders the editor. Since this should only be announced if the language is automatically set (since that is surprising). We should not announce in all other cases.

**Comment by TylerLeonhardt** [comment]
@isidorn, I was going to add the `alert` here:
https://github.com/microsoft/vscode/blob/c830429cb1da0b4e20d15a031f24c8f9f6e1460a/src/vs/workbench/common/editor/textEditorModel.ts#L121

but it looks like it violates imports:
![image](https://user-images.githubusercontent.com/2644648/130526659-f797f0ec-7bf1-4e12-942a-aa1b1846d28e.png)

I'm surprised the `alert` is under browser honestly. Could that be moved?

**Comment by isidorn** [comment]
@TylerLeonhardt `aria` is using browser specific approach to make a screen reader announce something. It is using `HTMLElements` and `aria-live` flags. Also `aria` stands for `Accessible Rich Internet Applications`, so it is very browser specific. I would not move it.

So I suggest to call it one layer above if possible.

**Comment by bpasero** [comment]
Alternatively we need to expose this from a service in `common` that is then implemented in `browser`?

**Comment by isidorn** [comment]
@bpasero yeah we could move it to a service, but I like the simplicity of just importing a helper method.

**Comment by TylerLeonhardt** [comment]
I have opened a PR here: https://github.com/microsoft/vscode/pull/131947

that creates a new service `AriaAlertService` with a single function: `alert`

Let me know what you think.

**Comment by TylerLeonhardt** [comment]
Verification:
* Enable a screen reader
* Open an untitled file
* Paste in a large enough snippet to trigger auto detect
* Listen for something like “the language has been changed”
